CC/Sounder/BaseRadioSetUHD.cc

```diff
@@ -190,28 +187,53 @@ void* BaseRadioSetUHD::init_launch(void* in_context) {
 }
 
 void BaseRadioSetUHD::init(BaseRadioContext* context) {
-  int i = context->tid;
   int c = context->cell;
   std::atomic_ulong* thread_count = context->thread_count;
   delete context;
 
   MLPD_TRACE("Deleting context for tid: %d\n", i);
 
-  auto channels = Utils::strToChannels(_cfg->bs_channel());
   std::map<std::string, std::string> args;
 
   args["driver"] = "uhd";
-  args["addr"] = _cfg->bs_sdr_ids().at(c).at(i);
-  std::cout << "Init bsRadios: " << args["addr"] << std::endl;
-  //    }
+  size_t num_radios = _cfg->n_bs_sdrs()[c];
+  std::vector<std::string> address_list;
+  for (size_t i = 0; i < num_radios; i++) {
+    std::ostringstream oss;
+    oss << "addr" << i;
+    address_list.push_back(oss.str());
+    args[address_list.at(i)] = _cfg->bs_sdr_ids().at(c).at(i);
+  }
+
   args["timeout"] = "1000000";
   try {
     bsRadios = nullptr;
```

Comment:
probably no reason to set this to null here.  Might make more sense to make sure it is null before you assign it with New